gloss/crt0.S

```diff
@@ -171,6 +192,11 @@ _skip_init:
   csrwi fcsr, 0
 1:
 
+  /* Set the real trap handler and use direct mode as default */
```

Comment:
By doing the following, we would be coupling the startup code with metal.
Although this would be the preference, there are cases already that some user want to use their own.
It is best to have the startup as minimal as possible, and allow the rest for the user to add their pieces.

Reply:
Yes, that is why I move out the mtvec setting from interrupt initial function. In original case, when user calling the interrupt initial function, it would set the mtvec to __metal_exception_handler, and causes the user's trap handler be overwrite.

Reply:
How did the user set the original, set the initial trap handler? which api was used?

Reply:
Before this patch, if user has own trap handler, they can set the mtvec value to their handler address in their program by using CSR instruction, after that, user call metal_interrupt_init to initialize interrupt controller, the mtvec would be set to  __metal_exception_handler by __metal_driver_riscv_cpu_controller_interrupt_init function.